### PR TITLE
chore: made dns inline with gcp and aws tfs

### PR DESCRIPTION
**Pull request by tomhobson**
I've been trying to bring this inline with: https://github.com/jenkins-x/terraform-google-jx

Before doing some future updates:
* Autoscaling
* Extra domains
* Premium tiers for container registries

### CODE REVIEW

File: jx-requirements.yml.tpl [hunk 1]
```diff
@@ -11,19 +11,15 @@ spec:
       secretStorage:
         keyVaultName: ${key_vault_name}
   %{ endif }
-  %{ if dns_enabled }
       dns:
         resourceGroup: ${dns_resource_group}
         tenantId: ${dns_tenant_id}
         subscriptionId: ${dns_subscription_id}
-  %{ endif }
```
Comment (hervelemeur) [inline jx-requirements.yml.tpl:19]:
```suggestion
  %{ if domain_enabled }
      dns:
        resourceGroup: ${dns_resource_group}
        tenantId: ${dns_tenant_id}
        subscriptionId: ${dns_subscription_id}
  %{ endif }
```

Not sure but this part seems to be only required when the domain is enabled.

File: jx-requirements.yml.tpl [hunk 2]
```diff
@@ -11,19 +11,15 @@ spec:
...
   environments:
     - key: dev
   ingress:
     domain: ${domain_name}
```
Comment (hervelemeur) [inline jx-requirements.yml.tpl:21]:
```suggestion
    domain: ${domain}
```

File: terraform-jx-azuredns/variables.tf [hunk 3]
```diff
@@ -1,33 +1,28 @@
...
   type        = string
-  description = "The resource group in which the Azure DNS apex domain resides. Required if apex_domain_integration_enabled is true"
+  description = "The Azure resource group the apex domain is managed by, used to write recordsets for a subdomain if set.  Defaults to current project."
   default     = ""
 }
-variable "resource_group_name" {
+variable "apex_domain_integration_enabled" {
+  type        = bool
+  default     = false
+  description = "If parent / apex domain is managed in the same "
```
[the remark this answers is not in the document]
Comment (hervelemeur) [reply terraform-jx-azuredns/variables.tf:21]: (I presume it's directly from the google terraform repo which has the same issue)

File: variables.tf [hunk 4]
```diff
@@ -107,22 +107,22 @@ variable "logging_retention_days" {
...
   type        = bool
-  default     = false
+  default     = true
   description = "Flag that when set attempts to create delegation records in apex domain to point to domain created by this module"
 }
-variable "dns_enabled" {
-  type        = bool
-  default     = false
-  description = "Flag that when set creates an Azure DNS zone for JX"
-}
```
Comment (hervelemeur) [inline variables.tf:117]:
```suggestion
variable "dns_enabled" {
  type        = bool
  default     = false
  description = "** Deprecated** Set apex_domain instead"
}
```

**Review by hervelemeur — CHANGES_REQUESTED** [review]

File: terraform-jx-azuredns/variables.tf (hunk 3 above)
[the remark this answers is not in the document]
Comment (tomhobson) [reply terraform-jx-azuredns/variables.tf:21]: Yeah I'll update this in both if you can decide what should be there?

@jstrachan @hervelemeur

File: jx-requirements.yml.tpl (hunk 1 above)
Comment (tomhobson) [reply jx-requirements.yml.tpl:19]: Looks like this isn't done within the gcp one: https://github.com/jenkins-x/terraform-google-jx/blob/master/modules/jx-requirements-v3.yml.tpl#L10

Can add it but it's up to you

File: terraform-jx-azuredns/variables.tf (hunk 3 above)
[the remark this answers is not in the document]
Comment (hervelemeur) [reply terraform-jx-azuredns/variables.tf:21]: Good question ^^ I think it should be "namespace" but I'm not sure.

File: jx-requirements.yml.tpl (hunk 1 above)
Comment (hervelemeur) [reply jx-requirements.yml.tpl:19]: Indeed, there isn't any "dns" entry in GCP jx-requirements.

But as there is one for Azure, I think we should keep it declared only when the domain is enabled. I don't know what would happen if it's declared without any domain enabled.

File: terraform-jx-azuredns/variables.tf (hunk 3 above)
[the remark this answers is not in the document]
Comment (tomhobson) [reply terraform-jx-azuredns/variables.tf]: I've taken the original value, seemed to be a good enough description

**Pull request merged**